commit 2436ce1bad9ec8756d2c6626b794ddac67cd86ef
Author: Dave Jiang <dave.jiang@intel.com>
Date:   Wed Sep 5 13:31:40 2018 -0700

    uaccess: Fix is_source param for check_copy_size() in copy_to_iter_mcsafe()
    
    commit dfb06cba8c73c0704710b2e3fbe2c35ac66a01b4 upstream.
    
    copy_to_iter_mcsafe() is passing in the is_source parameter as "false"
    to check_copy_size(). This is different than what copy_to_iter() does.
    Also, the addr parameter passed to check_copy_size() is the source so
    therefore we should be passing in "true" instead.
    
    Fixes: 8780356ef630 ("x86/asm/memcpy_mcsafe: Define copy_to_iter_mcsafe()")
    Cc: <stable@vger.kernel.org>
    Reported-by: Fan Du <fan.du@intel.com>
    Signed-off-by: Dave Jiang <dave.jiang@intel.com>
    Reviewed-by: Vishal Verma <vishal.l.verma@intel.com>
    Reported-by: Wenwei Tao <wenwei.tww@alibaba-inc.com>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/include/linux/uio.h b/include/linux/uio.h
index 409c845d4cd3..422b1c01ee0d 100644
--- a/include/linux/uio.h
+++ b/include/linux/uio.h
@@ -172,7 +172,7 @@ size_t copy_from_iter_flushcache(void *addr, size_t bytes, struct iov_iter *i)
 static __always_inline __must_check
 size_t copy_to_iter_mcsafe(void *addr, size_t bytes, struct iov_iter *i)
 {
-	if (unlikely(!check_copy_size(addr, bytes, false)))
+	if (unlikely(!check_copy_size(addr, bytes, true)))
 		return 0;
 	else
 		return _copy_to_iter_mcsafe(addr, bytes, i);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1245_2436ce1bad9ec8756d2c6626b794ddac67cd86ef.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1245_2436ce1bad9ec8756d2c6626b794ddac67cd86ef.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
